lxd/db/cluster/update.go

```diff
@@ -54,6 +55,32 @@ var updates = map[int]schema.Update{
 	17: updateFromV16,
 	18: updateFromV17,
 	19: updateFromV18,
+	20: updateFromV19,
+}
+
+// Add a new "arch" column to the "nodes" table.
+func updateFromV19(tx *sql.Tx) error {
+	// The column has a not-null constraint and a default value of
+	// 0. However, leaving the 0 default won't effectively be accepted when
+	// creating a new, due to the check constraint, so we are sure to end
+	// up with a valid value.
+	stmt := fmt.Sprintf(`
+ALTER TABLE nodes ADD COLUMN arch INTEGER NOT NULL DEFAULT 0 CHECK (arch >= %d AND arch <= %d)
```

Comment:
Hmm, I don't like that `CHECK` so much. Making sure it's > 0 on INSERT is fine but checking that it's not higher than the last arch means that adding an architecture will now need a schema change which seems a bit annoying.

Reply:
Yeah, I had mixed feelings too. I relaxed it to ```CHECK(arch > 0)```.